### ISSUE TITLE
Add more examples & documentation

### ISSUE THREAD

**Issue by robertmircea** [robertmircea/RateLimiters#2]

**Comment by Tsaukpaetra** [comment]
Yes please! Despite the variable names being mostly logical, I'm still having trouble selecting what type of bucket thing to use. For instance, what is the difference between stepInterval and stepIntervalInMilliseconds (both required for a StepUpLeakyTokenBucket)? I'm switching through three tabs in the source files to try and puzzle that answer out at the moment...
